modules/ROOT/edits/execution-engine.adoc

```diff
@@ -0,0 +1,273 @@
+----
+
+Notice that there’s a `poolStrategy` parameter. This parameter exists for backwards compatibility reasons and allows you to switch back to the three pools scheme from prior versions of mule:
+
+[source, xml, linenums]
+----
+<ee:scheduler-pools gracefulShutdownTimeout="15000">
+   <ee:cpu-light
+           poolSize="2"
+           queueSize="1024"/>
+   <ee:io
+           corePoolSize="1"
+           maxPoolSize="2"
+           queueSize="0"
+           keepAlive="30000"/>
+   <ee:cpu-intensive
+           poolSize="4"
+           queueSize="2048"/>
+</ee:scheduler-pools>
+----
+
+==== Considerations
+
+Applying pool configurations at the application level causes Mule to create a completely new set of thread pools for the Mule app. This configuration does not change the default settings configured in the `scheduler-conf.properties` file, which is particularly important for on-premises deployments because many Mule apps can be deployed to the same Mule instance.
+
+MuleSoft recommends running Mule using default settings.
+If you are upgrading from Mule 4.1.x or Mule 4.2.x you can use the DEDICATED strategy so that any customizations applied in the `scheduler-conf.properties` file can still be used. However, MuleSoft still recommends to try the new default setting first and then determine whether the optimizations are still required. Also, try first to customize the new UBER strategy before falling back to the legacy mode.
+
+In any case, MuleSoft recommends consulting with Support before going into production with customized settings.
+
+== Custom Pools
+
+Besides the unique UBER thread pool, some components might create additional pools for specific purposes, such as:
+
+* NIO Selectors +
+Enables non-blocking IO. Each connector can create as many as required.
+* Recurring tasks pools +
+Some connectors or components might create specific pools to perform recurring tasks (e.g: expiration monitors, queue consumers, etc.)
+
+[[backpressure]]
+== Back-pressure
+
+Under heavy load, Mule might not have resources available to handle a specific event. This issue might occur because all threads are busy and cannot perform the handoff of the newly arrived event or because the current flow’s `maxConcurrency` value has been exceeded already.
+
+In case Mule cannot handle an event, a message is logged about the condition: `Flow 'flowName' is unable to accept new events at this time`. Also, Mule sends a notification to the flow source to perform any required actions.
+
+The actions Mule performs on back-pressure are specific to each connector’s source. For instance, an `http:listener` might return a `503` error code, while a message broker listener might provide the option to either wait for resources to be available or drop the message. In some cases, a source might disconnect from a remote system to avoid getting more data than it can process and then reconnect once the server state is normalized.
+
+== Upgrading from Mule 4.2/4.1
```

Comment:
```suggestion
== Upgrading from Mule 4.2 or 4.1
```

Reply:
Does this section come too late; haven't we already referred to this earlier in this set?